packages/web-components/fast-foundation/src/calendar/calendar.ts

```diff
@@ -0,0 +1,238 @@
+import { attr, FASTElement } from "@microsoft/fast-element";
+
+/**
+ * enum representing the different month and weekday formats
+ * @public
+ */
+export type DateStyle = "long" | "narrow" | "short";
+
+/**
+ * Information about a month
+ * @public
+ */
+export type MonthInfo = {
+    month: number;
+    year: number;
+    length: number;
+    start: number;
+};
+
+/**
+ * Calendar information needed for rendering
+ * including the next and previous months
+ * @public
+ */
+export type CalendarInfo = MonthInfo & {
+    previous: MonthInfo;
+    next: MonthInfo;
+};
+
+/**
+ * Calendar component
+ * @public
+ */
+export class Calendar extends FASTElement {
+    /**
+     * Today's date
+     * @internal
+     */
+    private today: Date = new Date();
```

Comment:
If the calendar loads at 11:59PM and the user interacts at 12:00am, does this need to update to the next day?

Reply:
It should only be used during the rendering cycle. The only case I can see is if the calendar is updated, triggering a re-rendering. It would be working at a static Date during class construction. I've replaced it to use new Date() at their calls so that it will be updated on re-renders.